src/main/resources/com/google/api/codegen/ruby/Rakefile.snip

```diff
@@ -1,5 +1,76 @@
+@extends "ruby/common.snip"
+
 @snippet generate(metadata)
-  @# -*- ruby -*-
-  require 'bundler/gem_tasks'
+  {@header(metadata.fileHeader)}
+
+  {@unitTestsTask(metadata)}
+  @if smokeTestsTask(metadata)
+
+    {@smokeTestsTask(metadata)}
+  @else
+  @end
+@end
+
+@private header(fileHeader)
+  {@toComments(fileHeader.copyrightLines)}
+  #
+  {@toComments(fileHeader.licenseLines)}
+
+  require "bundler/setup"
+  require "bundler/gem_tasks"
+
+  require "rubocop/rake_task"
+  RuboCop::RakeTask.new
+@end
+
+@private unitTestsTask(metadata)
+  desc "Run tests."
+  task :test do
+    $LOAD_PATH.unshift "lib", "test"
+    Dir.glob("test/**/*_test.rb").each { |file| require_relative file }
+  end
+
+  namespace :test do
+    desc "Runs tests with coverage."
+    task :coverage do
+      require "simplecov"
+      SimpleCov.start do
+        command_name "{@metadata.identifier}"
+        track_files "lib/**/*.rb"
+        add_filter "test/"
+      end
+
+      Rake::Task["test"].invoke
+    end
+  end
+@end
+
+@private smokeTestsTask(metadata)
+  @if metadata.hasSmokeTests
+    desc "Runs the smoke tests."
+    task :smoke_test, :project do |t, args|
+      ENV["SMOKE_TEST_PROJECT"] = :project || ENV["SMOKE_TEST_PROJECT"]
+      if ENV["SMOKE_TEST_PROJECT"].nil?
+        fail "You must provide a project. e.g. rake smoke_test[test123] or SMOKE_TEST_PROJECT=test123 rake smoke_test"
+      end
+
+      $LOAD_PATH.unshift "lib", "smoke_test"
+      Dir.glob("smoke_test/**/*smoke_test.rb").each { |file| require_relative file }
+    end
+
+    namespace :smoke_test do
+      desc "Runs smoke tests with coverage."
+      task :coverage, :project do |t, args|
+        require "simplecov"
+        SimpleCov.start do
+          command_name "{@metadata.identifier}"
+          track_files "lib/**/*.rb"
+          add_filter "smoke_test/"
+        end
 
+        Rake::Task["smoke_test"].invoke
+      end
+    end
+  @else
+  @end
 @end
```

Comment:
It looks like the 'default' task is missing:

task :default => :test

Also it's nice to have a 'ci' task. Please refer to https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/google-cloud-vision/Rakefile